tls/s2n_psk.h

```diff
@@ -89,9 +69,24 @@ int s2n_psk_calculate_binder(struct s2n_psk *psk, const struct s2n_blob *binder_
 int s2n_psk_verify_binder(struct s2n_connection *conn, struct s2n_psk *psk,
         const struct s2n_blob *partial_client_hello, struct s2n_blob *binder_to_verify);
 
-typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn, 
+typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn,
                                           struct s2n_psk_identity *identities, size_t identities_length,
-                                          uint16_t *chosen_wire_index);                     
+                                          uint16_t *chosen_wire_index);
 /* This function will be labeled S2N_API and become a publicly visible api once we release the psk API. */
 int s2n_config_set_psk_selection_callback(struct s2n_connection *conn, s2n_psk_selection_callback cb);
 
+/* Public Interface -- will be made visible and moved to s2n.h when the PSK feature is released */
+
+struct s2n_psk;
+typedef enum {
+    S2N_PSK_HMAC_SHA224 = 0,
+    S2N_PSK_HMAC_SHA256,
+    S2N_PSK_HMAC_SHA384,
+} s2n_psk_hmac;
+
+struct s2n_psk* s2n_external_psk_new();
+int s2n_psk_free(struct s2n_psk **psk);
+int s2n_psk_set_identity(struct s2n_psk *psk, const uint8_t *identity, uint16_t identity_size);
+int s2n_psk_set_secret(struct s2n_psk *psk, const uint8_t *secret, uint16_t secret_size);
```

Comment:
Isn't `new` more apt to the behavior of these functions as opposed to `set`? The way I look at it: 
`_new` suggests that we are allocating new memory and `set` suggests we are setting to previously allocated memory.

Reply:
Idk, our APIs don't usually make that distinction. Most APIs are "set_X" even though we copy X into new memory. I worry "new_X" is more confusing / suggests we're doing something more special than just setting a blob.

Reply:
I prefer the `_new` notation here as we actually allocate new memory for the blob.data (realloc(b->data, size)). We should probably fixate on a distinction for the usage of `new` and `set` for other s2n functions as well (ofc as a backlog task and not part of this PR).

Reply:
I prefer the `_new` notation here as we actually allocate new memory for the blob.data `(realloc(b->data, size))`. We should probably fixate on a distinction for the usage of `new` and `set` for other s2n functions as well (ofc as a backlog task and not part of this PR).